dapps/src/endpoint.rs

```diff
@@ -42,5 +43,5 @@ pub type Handler = server::Handler<net::HttpStream> + Send;
 pub trait Endpoint : Send + Sync {
 	fn info(&self) -> Option<&EndpointInfo> { None }
 
-	fn to_handler(&self, path: EndpointPath) -> Box<Handler>;
+	fn to_handler(&self, path: EndpointPath, control: Option<Control>) -> Box<Handler>;
```

Comment:
The style `Option<hyper::Control>` might be more self-explanatory about what exactly a `Control` is controlling.